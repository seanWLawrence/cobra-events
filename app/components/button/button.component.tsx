import classNames from "classnames";
import { classed } from "@tw-classed/react";

import * as theme from "~/theme";

export const Button = classed(
  "button",
  classNames(
    theme.padding.formElement,
    theme.font.formElement,
    theme.text.formElement,
    theme.rounded,
    theme.activeState,
    theme.inactiveState.button,
    theme.border
  ),
  {
    variants: {
      color: {
        primary:
          "bg-indigo-600 text-white hover:bg-indigo-700 focus:bg-indigo-700",
        secondary:
          "bg-transparent text-indigo-600 hover:bg-indigo-50 focus:bg-indigo-50",
      },
      defaultVariants: {
        color: "primary",
      },
    },
  }
);

export type ButtonProps = React.ComponentProps<typeof Button>;
