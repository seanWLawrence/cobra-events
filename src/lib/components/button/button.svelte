<script lang="ts">
	import classNames from 'classnames';

	import * as theme from '$lib/theme';

	export let variant: 'primary' | 'secondary' = 'primary';

	const className = classNames(
		theme.padding.formElement,
		theme.font.formElement,
		theme.text.formElement,
		theme.rounded,
		theme.activeState,
		theme.inactiveState.button,
		theme.border,
		{
			'bg-indigo-600 text-white hover:bg-indigo-700 focus:bg-indigo-700': variant === 'primary',
			'bg-transparent text-indigo-600 hover:bg-indigo-50 focus:bg-indigo-50':
				variant === 'secondary'
		}
	);
</script>

<button class={className}><slot /></button>
