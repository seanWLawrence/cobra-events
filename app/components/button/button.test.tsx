import { expect, test } from "vitest";
import { render, screen } from "@testing-library/react";

import { Button } from "./button.component";

test("matches snapshot", () => {
  const name = "some name";

  render(<Button>{name}</Button>);
  const button = screen.getByRole("button", { name });

  expect(button).toMatchSnapshot();
});
