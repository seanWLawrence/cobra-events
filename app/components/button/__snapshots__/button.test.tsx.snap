// Vitest Snapshot v1

exports[`matches snapshot 1`] = `
<button
  class="py-4 px-6 font-mono text-2xl rounded-lg hover:shadow-none focus:shadow-none focus:outline-8 hover:outline-8 focus:outline-indigo-300 hover:outline-indigo-300 focus:outline-offset-4 hover:outline-offset-4 shadow-lg shadow-slate-300 border-2 border-indigo-600"
>
  some name
</button>
`;
